climakitae/util/utils.py

```diff
@@ -135,8 +141,17 @@ def get_closest_gridcell(data, lat, lon, print_coords=True):
     # Convert coordinates to x,y
     x, y = lat_lon_to_model_projection.transform(lon, lat)
 
-    # Get closest gridcell
-    closest_gridcell = data.sel(x=x, y=y, method="nearest")
+    # Get closest gridcell using tolerance
+    # If input point outside of dataset by greater than one
+    # grid cell, then None is returned
+    try:
+        closest_gridcell = data.sel(x=x, y=y, method="nearest", tolerance=tolerance)
+    except KeyError:
+        print(
+            "Input coordinates: (%.2f, %.2f)" % (lat, lon)
```

Comment:
nit: if you wanted to use an fstring here, you could shorten this, like `print(f"Input coordinates: ({lat0:.2f}, {lon0:.2f}) OUTSIDE .......")`.

Also would suggest slightly modifying the helper text, like "Input coords (x, y) are OUTSIDE of data extent by more than one cell. Returning None."

Reply:
Adopted in [419c0e2](https://github.com/cal-adapt/climakitae/pull/392/commits/419c0e27cd4b2d46666a4190b4b55d031ca942d6)